src/components/Buttons/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,43 @@
+import { SetStateAction } from "react";
+import style from "./Checkbox.module.scss";
+import { RegisterCheckboxValue } from "@/services/types";
+
+interface ICheckbox {
+  label: string;
+  initialValue: RegisterCheckboxValue;
+  registerName: string;
```

Comment:
**nit:** it's just a name. Focus on native input attributes. This is how this `register` works. It returns native attributes that match form state provided by react-hook-form: https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement